### PR TITLE
Removed logger overrides

### PR DESCRIPTION
**Pull request by cleentfaar**
see #29

afaik there is no reason to use the `DefaultLogger` at all here, either the user passes it's own logger or the `NoOpLogger` is used.

### CODE REVIEW

**Comment by cleentfaar** [comment]
FYI fixing tests now...

**Comment by cleentfaar** [comment]
I could not run tests separately because of the `DATAFILE` constant you defined in a non-autoloading place (globally). And there are a lot of classes that are only used to check good/bad arguments (e.g. `InvalidEventDispatcher` / `ValidEventDispatcher`); this can be easily mocked without the need of custom classes.

I don't have time to refactor everything for you but I was able to remove some off the useless stubs and moved the `DATAFILE` constant to an (autoloadable) class so tests can be run individually without any mandatory `includes(...)`.

File: tests/TestBucketer.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * limitations under the License.
+ */
+namespace Optimizely\Tests;
+
+use Optimizely\Bucketer;
+
+/**
+ * Class TestBucketer
+ * Extending Bucketer for the sake of tests.
+ * In PHP we cannot mock private/protected methods and so this was the most novel way to test.
```
Comment (cleentfaar) [inline tests/TestBucketer.php]: FYI (just my two cents): imo you should never need to test private methods, relying on this indicates your class or test has too many responsibilities (breaking SRP).

File: composer.json [hunk 2]
```diff
@@ -24,5 +24,10 @@
     "psr-4": {
       "": "src/"
     }
+  },
+  "autoload-dev": {
```
Comment (aliabbasrizvi) [inline composer.json]: Thank you for adding this.

**Comment by mikeng13** [comment]
Closing this due to unresponsiveness. If you'd like to submit your changes again, please open PR against the master branch.

**Pull request closed without merging**